akka-stream/src/main/scala/akka/stream/impl/StreamOfStreamProcessors.scala

```diff
@@ -82,9 +82,11 @@ private[akka] object MultiStreamOutputProcessor {
     private def closePublisher(withState: CompletedState): Unit = {
       subscriptionTimeout.cancel()
       state.getAndSet(withState) match {
-        case Attached(sub)     ⇒ closeSubscriber(sub, withState)
         case _: CompletedState ⇒ throw new IllegalStateException("Attempted to double shutdown publisher")
-        case Open              ⇒ // No action needed
+        case Attached(sub) ⇒
+          if (subscriber eq null) tryOnSubscribe(sub, CancelledSubscription)
```

Comment:
The fix here makes sense. 

I'm not quite sure in what scenario the actor would not stop in `closeSubscriber()`, AFAICS since state here is `Attached` we won't go into the `Open`state on the closeSubscriber, so the only path that does nothing to the actor is `case Failed(e: SpecViolation) ⇒ // nothing to do`.. Unless you mean that after receiving complete/error the actor is wrongly implemented and does not stop - but then it's his fault and not this code?